src/Microsoft.Health.Dicom.Api/Web/LimitStream.cs

```diff
@@ -0,0 +1,114 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Api.Web
+{
+    class LimitStream : Stream
```

Comment:
nit: visibility of classes should be explicit. This class is probably `internal`